### DISCUSSION TITLE
Domain Name

### DISCUSSION THREAD

**Discussion by elsycharles** [community/community#133934] [GitHub Education] [answered]
### Hi everyone,

Is there a way to have a custom domain for free in Student Pack of Github and without credit card ?

**Comment by Sivamani-18** [comment, marked as answer]
Yes, there are ways to get a custom domain for free with the GitHub Student Developer Pack and without needing a credit card. Here are some options included in the pack:

1. **Name.com**:
   - **Offer**: One free domain name registration for a year.
   - **Requirements**: Available for students through the GitHub Student Developer Pack.
   - **How to Get It**: 
     - Sign up for the GitHub Student Developer Pack [here](https://education.github.com/pack).
     - Once approved, navigate to the offers and select the Name.com offer to redeem your free domain.

2. **Namecheap**:
   - **Offer**: Free domain registration for a `.me` TLD for one year.
   - **Requirements**: Available through the GitHub Student Developer Pack.
   - **How to Get It**:
     - After getting the GitHub Student Developer Pack, go to the Namecheap offer.
     - Follow the instructions to claim your free domain.

These offers allow you to obtain a custom domain for free for one year. Here’s a brief guide on how to start:

1. **Sign up for the GitHub Student Developer Pack**:
   - Go to [GitHub Education](https://education.github.com/pack).
   - Apply for the Student Developer Pack by providing proof of your student status.

2. **Redeem the Domain Offer**:
   - Once you have access to the pack, look for the domain registration offers (from Name.com or Namecheap).
   - Follow the instructions provided by the specific domain registrar to claim your free domain.

**Reply by elsycharles** [reply]
Thanks you so much, I'll proceed that

**Reply by elsycharles** [reply]
I've tried the **Name.com** but it ask me credit card
![image](https://github.com/user-attachments/assets/3aef552f-309b-4358-a726-e7bccddd65e8)
So I'm trying the Namecheap ...

**Reply by Sivamani-18** [reply]
If Name.com requires a credit card and you wish to avoid using one, you can try other alternatives provided in the GitHub Student Developer Pack. Specifically, Namecheap often offers free domains to students without the need for a credit card.

**Reply by elsycharles** [reply]
I'm trying Namecheap now but it makes me verify my Student Status again so I have to wait the feed back of their google form 
![image](https://github.com/user-attachments/assets/ac45fcbc-e6ff-4eff-a0aa-d7d5dfce8adb)

**Reply by Sivamani-18** [reply]
Check this  boilerplate https://github.com/Sivamani-18/vite-react-ts-boilerplate

**Comment by ShenalSen** [comment]
Yes that is Namecheap. I am using that.

**Reply by elsycharles** [reply]
Thank you, that reassure.

**Comment by ShenalSen** [comment]
Hi @elsy-sc it looks like you are in wrong path. First of all you need to register the [GitHub Student Developer Pack page](https://education.github.com/pack). then register with your university details and upload valid document of your university (Your student ID, University offer letter or your students records page. It takes few seconds for review your application and then they will give you the feedback for that. If your application is approved , You'll receive an email confirmation and then they will give you the student developer pack benefits.
Under the student developer pack benefits you can see the [namecheap.com] (https://www.namecheap.com/) offer.  then you can simply register for name cheap.com and get free domain name for your website. And you can get more and more student benefits by registering student developer pack. (Github copilot , IntelliJ Idea Ultimate, Heroku, Microsoft azure clous services etc.) and you can earn the Github Pro badge.

**Reply by elsycharles** [reply]
I've done that but when I arrived in the namecheap education page this shows me that 
![image](https://github.com/user-attachments/assets/bd247538-9087-4a68-a216-8f858f392f11)

**Reply by ShenalSen** [reply]
So what is your university?

**Comment by ShenalSen** [comment]
You cant get free domain name from any other providers for your student developer pack only one solution is https://www.namecheap.com/
cuz i am the on of undergraduate as you and tried lot  to get that

**Reply by HISHAN03** [reply]
i got a free gihub domain from namecheep but where can i manage the domain like change the cname and stuff?

**Reply by ShenalSen** [reply]
Hey bro, if u got a domain from Namecheap (even if it was free), and u wanna link it to ur GitHub repo or just manage stuff like the CNAME records here’s what u gotta do:

First, log in to ur Namecheap account. Then on the left sidebar, click Domain List – that’s where all ur domains are listed. Find the domain u wanna manage and hit "Manage" next to it.

Now scroll a bit and go to the "Advanced DNS" tab. That’s where u’ll see all the DNS stuff like CNAME, A records, etc. From there, u can edit or add new records to point ur domain to GitHub.

If u’re linking it to a GitHub repo (like for GitHub Pages), u’ll need to:

Add a CNAME record like this if you’re using a subdomain (like www.yourdomain.com):
Host: www, Value: yourusername.github.io

Or add A records if u’re using the root domain (yourdomain.com) GitHub gives u 4 IPs to point to.

Also, make sure to add a file called CNAME in ur repo (just the domain name inside it), and then in the GitHub Pages settings, paste ur domain in the "Custom domain" section.

**Reply by HISHAN03** [reply]
Thanks manh but the issue is when I login to my namecheap account I don't find my purchased domain.
